libs/surfaces/mackie/surface.cc

```diff
@@ -1158,6 +1168,38 @@ Surface::redisplay (PBD::microseconds_t now, bool force)
 		}
 	}
 
+	/* iCON P1-M/V1-M color update: full RGB SysEx for all 8 strips */
+	if ( is_v1m || is_platformMp ) {
+		uint8_t rgb_bytes[24] = {0};  // 8 strips × 3 bytes (init black/off)
+		uint32_t byte_idx = 0;
+		uint32_t strip_idx = 0;
+
+		for (Strips::const_iterator s = strips.begin(); s != strips.end() && strip_idx < 8; ++s, ++strip_idx) {
```

Comment:
Perhaps using a modern loop like
```suggestion
		for (const auto& s:  strips) {
			if  (strip_idx++ >= 8)
				break;
```
? Too bad C++20 ranges cannot be used to simplify more...